### DISCUSSION TITLE
Cacheing common packages

### DISCUSSION THREAD

**Discussion by artooras** [vercel/next.js#35591] [Help] [answered]
In my Next.js application I am observing that all the common js bundles, such as `main`, `_app`, etc. have some id attached to it - e.g. `_app.js?ts=1648142778564`. If I refresh the page, the id changes. I assume this is why these bundles are never cached as opposed to other bundles - for instance, I use `rollbar`, and `rollbar.min.js` shows as cached.

Is there a way to manage this cacheing behaviour in Next.js? I know there are cache controls [for individual routes](https://nextjs.org/docs/going-to-production#caching), but can I do it for the shared bundles?

**Comment by icyJoseph** [comment, marked as answer]
Interesting, do you have a repository to take a look? On my site, the `_app` and `_main` bundles are properly cached for 1 year. Do you maybe have `getInitialProps` on `_app` or something like that? Still strange.

**Reply by artooras** [reply]
Hm... I have a `getInitialProps` in my `_document.js` in order to enable SSR for `styled-components` following [this example](https://github.com/vercel/next.js/blob/main/examples/with-styled-components/pages/_document.js). Do you think that might be the reason?

**Reply by icyJoseph** [reply]
I don't think so, because I have sites that use styled components, and still, the bundles are cached for long. This is a bit strange, do you have custom webpack configs? My `_app` for example is requested as: `_next/static/chunks/pages/_app-304fbf46f1850281.js`, but yours seems to have a `timestamp` which is from `next dev`, are you seeing this in production with `next build` and `next start`? If it's on `next dev` then I think it is expected behaviour.

**Reply by artooras** [reply]
Oh dear... It's not the first time I'm caught out by this `prod` vs. `dev` thing. When will I learn... :) Thanks @icyJoseph, it seems cacheing works in prod.

On the other hand, in dev tools there's a checkbox to `disable cache` which is checked by default. Is this always user controlled? Or can this be set by the app/website being visited?

**Reply by icyJoseph** [reply]
It happens :) Don't forget to mark as answer, who knows, we may come back to the issue later on.

And yeah, from the Dev Tools you can monkey around as much as you want, including disabling any caching. I have it on when developing to get a feel for first time visit, but even if it's clicked to `disable cache`, the moment you close the Dev Tools it goes back to normal behaviour.

**Reply by artooras** [reply]
Thanks @icyJoseph!
